core-bundle/src/EventListener/PreviewUrlCreateListener.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\ArticleModel;
+use Contao\CoreBundle\Event\PreviewUrlCreateEvent;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\PageModel;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * @internal
+ */
+class PreviewUrlCreateListener
+{
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var ContaoFramework
+     */
+    private $framework;
+
+    public function __construct(RequestStack $requestStack, ContaoFramework $framework)
+    {
+        $this->requestStack = $requestStack;
+        $this->framework = $framework;
+    }
+
+    /**
+     * Adds the page ID to the front end preview URL.
+     *
+     * @throws \RuntimeException
+     */
+    public function __invoke(PreviewUrlCreateEvent $event): void
+    {
+        if (!$this->framework->isInitialized() || !($id = $event->getId()) || !\in_array($event->getKey(), ['page', 'article'], true)) {
```

Comment:
```suggestion
        if (!$this->framework->isInitialized() || (!$id = $event->getId()) || !\in_array($event->getKey(), ['page', 'article'], true)) {
```

This should be changed as well I guess?

See https://github.com/contao/contao/pull/3416/files#r709073556

Reply:
Yes please.